File: itv/SergioDeIscar-itv-java/SergioDeIscar-itv-java/src/main/java/org/example/exceptions/ItvExceptions.java
package org.example.exceptions;

public abstract class ItvExceptions extends RuntimeException{
    public ItvExceptions(String message) {
        super(message);
    }
    public static final class ItvMatriculaException extends ItvExceptions {
        public ItvMatriculaException() {
            super("Error -> Matricula no valida");
        }
    }

    public static final class ItvKilometroException extends ItvExceptions {
        public ItvKilometroException() {
            super("Error -> Kilometro no valida");
        }
    }

    public static final class ItvAnioException extends ItvExceptions {
        public ItvAnioException() {
            super("Error -> Anio de matriculacion no valido");
        }
    }

    public static final class ItvNumPuertasException extends ItvExceptions {
        public ItvNumPuertasException() {
            super("Error -> Numero de plazas no valido");
        }
    }

    public static final class ItvCilindradaException extends ItvExceptions {
        public ItvCilindradaException() {
            super("Error -> Cilindrada no valida");
        }
    }
}

File: itv/SergioDeIscar-itv-kotlin/SergioDeIscar-itv-kotlin/src/main/kotlin/exceptions/ItvExceptions.kt
package exceptions

sealed class ItvExceptions(message: String): RuntimeException(message)
class ItvMatriculaException: ItvExceptions("Error -> Matricula no valida")
class ItvKilometroException: ItvExceptions("Error -> Kilometro no valido")
class ItvAnioException: ItvExceptions("Error -> Anio de matriculacion no valido")

class ItvPlazasException: ItvExceptions("Error -> Numero de plazas no valido")
class ItvCilindradaException: ItvExceptions("Error -> Cilindrada no valida")